### ISSUE TITLE
My Home: remove 'edit your menu' checklist item 

### ISSUE THREAD

**Issue by rickybanister** [Automattic/wp-calypso#47334]
The checklist current doesn't do a great job of easing people into our product. 

<img width="1072" alt="Screenshot 2021-02-04 at 11 13 21" src="https://user-images.githubusercontent.com/87168/106870702-153aa280-66da-11eb-9b8a-4d0b63dc12ab.png">

Early on we expose people to multiple environments like the customizer, editor, and settings. Some settings could be intuited by default to let customers slowly learn where things live.

One of these is the menu editing UI in customizer. We can enable the 'add new pages to my menu' setting by default ( #46703 ) and remove this checklist item. When a customer has gotten the hang of adding pages and notices they'd like to make a change to their menu, that's a better time to introduce the customizer.

As @davipontesblog shares in a comment on the thread linked above, we could introduce a tooltip to the menus UI inviting customers to turn off this checkmark if they're going to make manual edits to their menu. 

We can also explore other ways to teach customers about menus via customer home, but it doesn't feel appropriate to do so within their first session, **therefore we should remove this checklist item**.

**Comment by xpurichan** [comment]
I'll close this report in favor of #49706 since it's being discussed there.

**Comment by simison** [comment]
@xpurichan Let's have this open — it's mentioned there but it's still significantly different design decision to swap list order vs remove one item.

Thanks for triaging! :-)

**Comment by xpurichan** [comment]
I can understand that, and thank you for the feedback!

**Comment by p-jackson** [comment]
Makes sense. The task was already being hidden for the Blank Canvas theme and when using full-site editing. I agree the customiser is something we should be encouraging the user to check out much further down the track.

(for reference, the task is currently called "Edit the site menu"

Patch that removes the task: D60892-code
Follow up PR to remove the client-side code for this task: #52493

With my change I've opt'd to do the simple thing of removing the task for everyone. Alternatively we could only show it for old sites, or for sites where the task has already been completed (so that we aren't modifying a list the user might assume is static). These are doable, but I don't think it's worth the extra code.

**Comment by simison** [comment]
Design discussion concluded that It would do more harm than good because it's basically impossible to know where to edit menus otherwise. Hence we only swapped its position lower recently.

We initially thought the requirement would be to add all new pages to menu by default (i.e. switch the setting for that on by default). HE feedback noted that being more confusing than helpful.

But would be cool to remove it — just needs an alternative approach for guidance. Let's ask @ollierozdarz for thoughts. :-)

Blank canvas doesn't have menus so the task was removed for it.

**Comment by autumnfjeld** [comment]
FYI related: changing the copy to "Edit the site menu": https://github.com/Automattic/wp-calypso/pull/52077 was shipped a couple weeks ago

**Comment by p-jackson** [comment]
> Blank canvas doesn't have menus so the task was removed for it.

I wonder if there's a way to do this generally, like remove it for all themes that don't have a menu. So that it doesn't show for e.g. Pollard

**Comment by p-jackson** [comment]
@jeffgolenski just another data point to pile on the list of things to consider. In particular this sentence I think:

> Design discussion concluded that It would do more harm than good because it's basically impossible to know where to edit menus otherwise. Hence we only swapped its position lower recently.

Having a task that drops users in the customiser doesn't feel like something we want, but also this ^^^

**Comment by jordesign** [comment]
> But would be cool to remove it — just needs an alternative approach for guidance.

Agreed we need alternative guidance - adding and editing menus is a very high frequency question for HEs and something that is really unclear for users _unless_ we point them there somehow.

> I wonder if there's a way to do this generally, like remove it for all themes that don't have a menu. So that it doesn't show for e.g. Pollard

If we can have items be conditional like that - it seems like a BIG win

> Having a task that drops users in the customiser doesn't feel like something we want

With Nav Unification - we now have Appearance > Menus _much_ more visible now. Perhaps we could be pointing user's there instead of the Customizer?

**Comment by ianstewart** [comment]
> Perhaps we could be pointing user's there instead of the Customizer?

+1. We may turn on the new navigation screen from core later as well.

> But would be cool to remove it — just needs an alternative approach for guidance.

Menus are really complex. The view tutorial link is really small. Perhaps the this checklist item (or most of them? or all of them?) should get users to view a tutorial instead of dumping them into confusing UI with complex mental models. Tutorials or tutorial completion events could be followed with a try it yourself link, perhaps. But a v1 could just offer users a way home.

**Comment by p-jackson** [comment]
> With Nav Unification - we now have Appearance > Menus much more visible now. Perhaps we could be pointing user's there instead of the Customizer?

Love it @jordesign! If it's quick to do it could be worth creating a PR just to see what it feels like.

**Comment by jordesign** [comment]
I think so. 

As a total aside - this kind of small PR is _exactly_ the kind of thing I want to see how it's done (as a path to better understanding Calypso dev)

**Comment by jeffgolenski** [comment]
Just jumping in @p-jackson @jordesign! 

Love your thoughts here! While getting to the menu editor is fairly straight forward, I tend to agree with @ianstewart. The menu system itself is fairly complex and of course dumping people right into the customizer is another mental barrier that's been mentioned in this thread! 

Short-term idea: I'm wondering if we could also entertain the idea of shifting the CTAs around a bit to direct folks to the tutorial first. Do we have customer insight data on this page? https://wordpress.com/support/menus/ I'd be curious to see what customers think of the quality of this tutorial.

Current CTA structure: 
![Screen Shot 2021-05-10 at 10 22 30 AM](https://user-images.githubusercontent.com/214813/117675032-34821980-b17a-11eb-91c3-53a776980e24.png)

You'll notice that the "add a menu" checklist task is the only one with a "skip for now" link next to the primary CTA. I don't think that's necessary since folks can use the nav on the right side to skip it anyway.

What if we did this and began keeping an eye on how folks are using it?

![Screen Shot 2021-05-10 at 10 29 05 AM](https://user-images.githubusercontent.com/214813/117675540-a78b9000-b17a-11eb-95d6-68ad8fee2492.png)

**Comment by jordesign** [comment]
I like this idea - taking people through the doc seems like a good plan.

**Comment by ktyfuller604** [comment]
> Do we have customer insight data on this page? https://wordpress.com/support/menus/ I'd be curious to see what customers think of the quality of this tutorial.

* From the survey at the bottom of the doc (albeit with a low take rate), the majority (around 70%) of the users found the doc helpful. Those that didn't find the guide helpful provided good feedback that the docs guild will integrate into the doc as part of our standard operating procedure for handling the feedback. 
* As part of a separate project, I'm on a team is working on creating a new video for this specific doc, so that will make it even more helpful (and could potentially be used on the My Home checklist.)
* According to Google Analytics, the majority of the traffic to the guide comes from Google searches (out of 2,086 page views last week, 64% came from a Google search)
* According to Tracks, very few users who visit this doc click on the Contact Support link (less than 1%)

**Comment by jordesign** [comment]
> As part of a separate project, I'm on a team is working on creating a new video for this specific doc, so that will make it even more helpful (and could potentially be used on the My Home checklist.)

This could be really good!

**Comment by p-jackson** [comment]
PR ready for review: #52990
Commenting here because this is where all the subscribers are and this could probably do with some feedback since it looks a bit gnarly.

<img width="951" alt="Screenshot 2021-05-19 at 3 30 28 PM" src="https://user-images.githubusercontent.com/1500769/118753131-7288e700-b8b8-11eb-8d8f-d6dfc45ca55e.png">

> You'll notice that the "add a menu" checklist task is the only one with a "skip for now" link next to the primary CTA

For the record, the mobile app task is also skippable. And I think it probably important for editing the menu to be skippable too, otherwise there's no way for the user to ever move past the site setup checklist and see the My Home views that lie beyond. And I don't think we should force every user to do the task since not every site will want a menu.

@jeffgolenski if we _really_ don't want the "skip" button then I think the task needs to be much easier to complete. Currently it's not marked complete until some menu setting is saved, but we could change it to be trivial to complete (e.g. clicking the support link completes the task, although this could also make the UI a bit jump as dismissing the support modal might auto-move the user to the next task).

I've left the tracks events unchanged: the "Add a menu" button sends the same event as if it were still the primary action for the task, and the support link sends the same event it did when it was rendered as part of the paragraph text.

**Comment by simison** [comment]
We could also rename "skip" to something else like "done"?

**Comment by autumnfjeld** [comment]
[Approved](https://github.com/Automattic/wp-calypso/pull/52990#pullrequestreview-663799816) from functionality point of view.   I kinda like "Done for now", but also happy to ship this improvement and wait for "further instructions" 😄  for another iteration.

**Comment by jeffgolenski** [comment]
> For the record, the mobile app task is also skippable. And I think it probably important for editing the menu to be skippable too, otherwise there's no way for the user to ever move past the site setup checklist and see the My Home views that lie beyond. And I don't think we should force every user to do the task since not every site will want a menu.

Understood! @p-jackson @autumnfjeld I'm not in love with having three links there, but here's something to consider!

**From a functionality stand-point**, "skip for now" serves no real purpose within this checklist. Customers can navigate the items on the right in a non-linear fashion of their choosing. The moment they click "Launch your site" the checklist disappears for them.

Here's a quick video:


https://user-images.githubusercontent.com/214813/119009963-9eac7280-b961-11eb-8452-700902c26a4d.mov



**From a cognitive stand-point**, "skip for now" implies that they may not be able to launch their site until that task is somehow completed. 

There's a deeper problem to explore here around setting proper expectations for launch, which I'll explore during medium-term work.

tl;dr - I think we should leave out any "skip for now" links since those serve linear onboarding patterns, but we don't have that here. I like @simison's idea of "Done"

**Comment by p-jackson** [comment]
You're right the setup does disappear post launch, but it's easily brought back by clicking "Show site setup".

https://user-images.githubusercontent.com/1500769/119058888-55bcf280-ba23-11eb-80ca-3233accc162f.mov

I also tried refreshing the page in the above video to see whether My Home would go back to displaying the congratulatory message, but it didn't.

> From a cognitive stand-point, "skip for now" implies that they may not be able to launch their site until that task is somehow completed.

I think you're right, "skip" isn't the right mental model. Maybe something like a cross in the upper-right is more what we're going for? It generally means "make what I'm looking at go away in a safe way"

<img width="951" alt="withcross" src="https://user-images.githubusercontent.com/1500769/119059084-b815f300-ba23-11eb-8196-642925fbb937.png">

This is probably part of the medium-term work you mention.

> tl;dr - I think we should leave out any "skip for now" links since those serve linear onboarding patterns, but we don't have that here.

I'll update the PR to remove the skip and try it out. Maybe there's a way to work around the issue of being stuck in site setup forever. Maybe we could special case this task so that it isn't required for completing site setup?

**Comment by p-jackson** [comment]
I've updated the PR to remove the skip button, but have also added logic so that the edit site menu task doesn't count towards completion of the list. Notice in the video below that we start with every task completed except editing the site menu and launching:

https://user-images.githubusercontent.com/1500769/119067772-e18c4a00-ba36-11eb-973a-4751c45da161.mov

The difference between this video and the one at https://github.com/Automattic/wp-calypso/issues/47334#issuecomment-845245100 is that the "You launched your site!" congratulatory message says "Show me what's next" instead of "Show site setup"

I think this implements the behaviour we've discussed here. I am still worried about users wanting to either complete (or "skip" or "tick off") each of the tasks. The green ticks of the list can be a strong incentive for completionists like me 😄 
If we go this direction I think we should let HEs know how to help users who are motivated to do all steps (which is to tell the users that the menu item isn't necessary)

**Comment by jordesign** [comment]
> The difference between this video and the one at #47334 (comment) is that the "You launched your site!" congratulatory message says "Show me what's next" instead of "Show site setup"

This feels more natural to me as a transition post-launch.

> I am still worried about users wanting to either complete (or "skip" or "tick off") each of the tasks. The green ticks of the list can be a strong incentive for completionists like me 😄
If we go this direction I think we should let HEs know how to help users who are motivated to do all steps (which is to tell the users that the menu item isn't necessary)

This *is* a behavior we witness sometimes. HEs can definitely confirm the step isn't needed. Is your worry that people won't launch the site until *after* that step is completed?

**Comment by p-jackson** [comment]
> Is your worry that people won't launch the site until _after_ that step is completed?

Yes, although my feeling probably isn't as strong as "won't". More like "people will be marginally disincentived from launching". But then you could argue that any step placed before launching will marginally decrease launch rate 🤷 Perhaps this is the best of both worlds until the day we can simply remove this difficult to complete task from the setup checklist.

**Comment by jeffgolenski** [comment]
Good stuff @p-jackson — thanks for that added work. Really like the "show me what's next" to push the forward motion of the flow. 

![Screen Shot 2021-05-24 at 3 26 13 PM](https://user-images.githubusercontent.com/214813/119397639-6b8d1a80-bca4-11eb-96b5-d2f212fcee26.png)


Do we happen to have data on how many people were launching, and then clicking "Show site setup" & "skip site setup" before? Seemed like a big hurdle to get around since the primary CTA took them a step back. 

I think we should also see a bump in the clicks on the subsequent task CTA items now. Let's keep an eye on those metrics as well!  cc @autumnfjeld

**Comment by p-jackson** [comment]
> Do we happen to have data on how many people were launching, and then clicking "Show site setup" & "skip site setup" before? Seemed like a big hurdle to get around since the primary CTA took them a step back.

Surprisingly we do not have that data (at least easily accessable). Adding in #53134.

**Comment by autumnfjeld** [comment]
Just a lil summary at this point in the thread discussion.  
This thread has led to some great, albeit complex discussions, which have exposed a number of user facing problems:
   * Menu Editing is confusing and it is difficult for us to even decide how to present that to users (to be or not to be in the site checklist?)  .  Also to be noted that some of this pain will be alleviated with FSE.  
   * Skipping things in the site checklist (Edit menu and mobile app) is counterintuitive to a checklist and maybe these items should be place somewhere else?
   * The "You launched your site!" banner is odd in that the primary CTA takes the user back to the site checklist!  That seems unhelpful and definitely not good for keeping the user moving forward. 

These problems require broader thinking, as has been mentioned in the comments above, and are part of the upcoming [medium-goals thinking](d2qGl-p2#medium-term-goals)  that @jeffgolenski is getting started with.   For me, that means those issues are being addressed in the context of user flow and UX and perhaps best place for continued discussion will be in the [My Home pod](d2qGl-p2) posts. :) 


For an interim solution, [Phil's PR](https://github.com/Automattic/wp-calypso/pull/52990) contains the following small improvements, [as shown in Phil's video comment](https://github.com/Automattic/wp-calypso/issues/47334#issuecomment-845585733)

1.  Update to "Edit the site menu" card to make "View tutorial" the primary CTA 
![image](https://user-images.githubusercontent.com/5665959/119431964-d5fe9300-bd56-11eb-8a6e-c786384aaced.png)


2.  Update to the way the the site checklist is considered "complete", such that the  “edit menu” task is programmatically labeled as skipped and so does not need to be completed by the user.  Thus after launch the user will see the following  even if they have not completed the “edit menu” task:
![image](https://user-images.githubusercontent.com/5665959/119431316-b61a9f80-bd55-11eb-98f1-4d3e95025e3d.png)


 Again, Phil's PR isn't meant to address all the problems discussed in this thread, but rather be an improvement we can ship now, while deeper thought goes into the larger flow and purpose of the checklist and post-launch experience.

**Comment by p-jackson** [comment]
The changes to the "edit your menu" task discussed here have shipped: #52990
It doesn't address every problem with editing menus, or even the original task as described by the title ("remove 'edit your menu' checklist item"), but it has implemented the changes as discussed here. So I think it's right to close this issue now.
